templates/tftmpl/root.go

```diff
@@ -41,11 +41,15 @@ const (
 	// variable is written to.
 	TFVarsFilename = "terraform.tfvars"
 
+	// VarsTFVarsFileName is the file name for a tfvars file which is generated and contains
+	// variables provided as part of the task configuration
+	VarsTFVarsFileName = "variables.tfvars"
```

Comment:
What's the context behind "variables.tfvars" filename? Could the filename be preserved from what was provided from the config/payload for debugging purposes?

Reply:
If we want to support multiple variables files in the working directory then we can do that, otherwise the name came from `variables.module.tf` and `variables.tf` which are related to this file.

Reply:
What's the thought behind collapsing user input variables into one file?

Reply:
So for TFC, the variables for all files just get loaded into a single list of parsed key values, and this list gets fed to TFC. My thought was to mirror this behaviour, where instead of feeding variables to the terraform module, I instead took that same list of variables and wrote a file to the working space. This file would then be used as input to the terraform driver, much in the same way the list is used as input to the TFC driver

Reply:
Cool thank for sharing. A single file maybe simpler to manage for future task updates too

Reply:
Not sure if this was already considered, would the `*auto.tfvars` naming scheme be useful for this feature? https://www.terraform.io/docs/language/values/variables.html#variable-definition-precedence

Reply:
I hadn't come across this, I can add the *.auto* to the tfvars file if that's the convention.